File: docs/build-html.py
#!/usr/bin/env python

import glob, os.path

pandoc_cmd  = '''pandoc -s --toc -N --css=styles.css -H _head-inc -B _nav-header -A _nav-footer -o %s %s'''
pandoc_cmd2 = '''pandoc -s          --css=styles.css -H _head-inc -B _nav-header -A _nav-footer -o %s %s'''

txt_files = glob.glob('*.txt')

for txt_fn in txt_files:
    html_fn = txt_fn[:-3] + 'html'
    if not os.path.exists(html_fn) or os.path.getmtime(txt_fn) > os.path.getmtime(html_fn):
        if txt_fn == 'index.txt' or txt_fn == 'appendix-a-index.txt' or txt_fn == 'appendix-g-glossary.txt':
            print "Processing:", txt_fn, '-->', html_fn
            os.system(pandoc_cmd2 % (html_fn, txt_fn))
        else:
            print "Processing:", txt_fn, '-->', html_fn
            os.system(pandoc_cmd % (html_fn, txt_fn))

File: docs/build-html.pl
#!/usr/bin/perl -w
use strict;
use v5.10;

my $prefix = shift(@ARGV) // '.';

my $pandoc_cmd = {
    normal => 'pandoc -s --toc --css=styles.css -H _head-inc -B _nav-header -A _nav-footer -o %s %s',
    'appendix-a-index.txt'    => 'pandoc -s   --css=styles.css -H _head-inc -B _nav-header -A _nav-footer -o %s %s',
    'appendix-g-glossary.txt' => 'pandoc -s   --css=styles.css -H _head-inc -B _nav-header -A _nav-footer -o %s %s',
    'index.txt' =>      'pandoc -s   --css=styles.css -H _head-inc                               -o %s %s', # -N
};

for my $txt_file (<*.txt>) {
    my $html_file = substr($txt_file,0,-3) . 'html';
    my $cmd = $pandoc_cmd->{ $txt_file } // $pandoc_cmd->{normal};
    #print "Processing:", $txt_file, '-->', $html_file
    system sprintf $cmd, "$prefix/$html_file", $txt_file;
}
